symmer/chemistry/fermionic_ham.py

```diff
@@ -267,9 +267,12 @@ def manual_T2_mp2(self) -> FermionOperator:
 
     @cached_property
     def hf_fermionic_basis_state(self):
-        hf_comp_basis_state = np.zeros(self.n_qubits, dtype=int)
-        hf_comp_basis_state[:self.n_electrons] = 1
-        return hf_comp_basis_state
+        n_alpha, n_beta = self.scf_method.nelec
+        hf_array = np.zeros(self.n_qubits)
+        hf_array[::2] = np.hstack([np.ones(n_alpha), np.zeros(self.n_qubits//2-n_alpha)])
```

Comment:
Hartree-Fock state was not being constructed correctly before - we filled the up/down orbitals equally, whereas this should be done according to the number of up/down electrons. For singlets these are the same, but leads to issues for non-singlet systems.